proxy/src/denim_routes.rs

```diff
@@ -1,16 +1,112 @@
-use denim_sam_common::denim_message::DeniableMessage;
+use denim_sam_common::{
+    denim_message::{deniable_message::MessageKind, DeniableMessage, KeyResponse},
+    Seed,
+};
+
 use sam_common::AccountId;
+use sam_server::managers::traits::account_manager::AccountManager;
 
 use crate::{
-    error::DenimRouterError,
-    managers::default::ClientRequest,
+    error::{DenimRouterError, LogicError},
+    logic::keys::{get_keys_for, update_seed},
+    managers::{default::ClientRequest, error::DenimKeyManagerError, traits::KeyRequestManager},
     state::{DenimState, StateType},
 };
 
 pub async fn denim_router<T: StateType>(
-    _state: DenimState<T>,
-    _request: ClientRequest,
-    _account_id: AccountId,
-) -> Result<DeniableMessage, DenimRouterError> {
-    todo!("Denim Proxy does not yet support denim routing");
+    state: &mut DenimState<T>,
+    request: ClientRequest,
+    account_id: AccountId,
+) -> Result<(), DenimRouterError> {
+    match request {
+        ClientRequest::BlockRequest(_, _block_request) => todo!(),
+        ClientRequest::KeyRequest(msg_id, key_request) => {
+            let requested_account_id = AccountId::try_from(key_request.account_id)
+                .map_err(|_| DenimRouterError::KeyRequestMalformed)?;
+
+            let key_bundle = match get_keys_for(
```

Comment:
please handle each case in a separate function to avoid 🗿

Reply:
wut the heel is 🗿🗿🗿🗿🗿🗿🗿